libglusterfs/src/inode.c

```diff
@@ -815,14 +831,21 @@ inode_grep(inode_table_t *table, inode_t *parent, const char *name)
     pthread_mutex_lock(&table->lock);
     {
         dentry = __dentry_grep(table, parent, name, hash);
-        if (dentry) {
-            inode = dentry->inode;
-            if (inode)
-                __inode_ref(inode, false);
-        }
     }
     pthread_mutex_unlock(&table->lock);
 
+    if (dentry->inode) {
```

Comment:
That's a bad idea. Once we leave the mutex, the `dentry` could have been destroyed by another thread, so we are accessing potentially invalid memory here. We cannot leave the mutex before taking a reference of the inode.